AmplifyPlugins/Storage/AWSS3StoragePluginFunctionalTests/AWSS3StoragePluginAccessLevelTests.swift

```diff
@@ -19,8 +20,8 @@ class AWSS3StoragePluginAccessLevelTests: AWSS3StoragePluginTestBase {
 
     // This is a run once function to set up users then use console to verify and run rest of these tests.
     func testSetUpOnce() {
-        signUpUser(username: user1)
-        signUpUser(username: user2)
+        AuthHelper.signUpUser(username: user1, password: password)
```

Comment:
We need to move the password out of the source file and into a setup run from the CLI. As it stands, if an attacker obtains our user pool ID (which remember is not considered sensitive), they could use the test storage bucket to store and distribute files.

Reply:
depending on what the attacker obtains, couldn't they also just create their own user as well? I thought that if `awsconfiguration.json` and `amplifyconfiguration.json` is never exposed then we are good, and the username and password for this user is specific for the user pool and wouldn't provide any value to the attacker.


if we need to move this out, i can follow a similar pattern and have another file like `credentials.json` that gets pulled down like https://github.com/aws-amplify/aws-sdk-ios/blob/f3967e2e2a648e8a114ef5dd50bef5e075668693/AWSAuthSDK/AWSAuthSDKTestAppUITests/AWSDropInUIUserPoolTests.swift#L24

Reply:
We treat things like pool IDs with different sensitivity than credentials like password. E.g., if we log a user pool ID, the config file doesn't need to be exposed. If this class is setting up tests that allow guest access, we need to protect the credentials. A `credentials.json` pattern or similar would be one way to do that, but let's talk about it IRL to make sure we don't end up with a catch-all for "everything that doesn't fit in a config file." :)

Reply:
discussed with tim and in the current state of things, we'll do a couple of things to make things more secure and less prone to abuse:
1. credentials.json to contain user/pass. this can be stored in the same pattern as the configuration files, ie. `AWSS3StorageIntegrationTests-credentials.json`. 
2. Not as urgent for API plugin but will do the same
3. add `*credentials.json` to .gitignore 
4. bucket policy updated to expire asap, looks like 1 day is the shortest possible.
5. The tests need to be prone to parallel execution. S3 keys defined in the tests should contain some unique identifier. there are many cases where the tests may be run in parallel. Multiple circleCI runs and developer using test account locally (though ideally developer should create their own resources using the readme instructions).

Reply:
<img width="691" alt="ExpireObjectsRule" src="https://user-images.githubusercontent.com/1365977/71014555-f8db4100-20a6-11ea-963c-8b0846d65b91.png">